### ISSUE TITLE
ExpoKit Android does not work with react-native-video library

### ISSUE THREAD

**Issue by cheolhow** [expo/expo#2298]
### Environment

Environment:
  OS: macOS High Sierra 10.13.5
  Node: 8.11.2
  Yarn: 1.5.1
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.4.1 Build version 9F2000
  Android Studio: 3.1 AI-173.4907809

Packages: (wanted => installed)
  expo: ^30.0.1 => 30.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz => 0.55.4

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/cheolho-743702a1-c2a8-4b07-a688-045da73c8cc0.tar.gz

app's target : expokit

### Steps to Reproduce

1. make a project expo init
2. expo eject with expokit
3. npm install -S react-native-video, react-native link react-native-video
4. open android studio, sync gradle and build
5. build failed (cannot find symbol class JavaNetCookieJar)
![image](https://user-images.githubusercontent.com/25603134/46120049-f04fde00-c248-11e8-894e-19fb3c6487e9.png)

 I am developing a project using react-native-video. It seems that expolib_v1.okhttp3 overwrote okhttp3 in the library. This seems to be a problem because there is no JavaNetCookieJar in expolib_v1.okhttp3.
 Build the same project in xcode and run normally with the iOS emulator.

**Comment by esamelson** [comment]
Hi @cheolhow . I'm not quite sure why this isn't working -- the `expolib_v1` version should not be overwriting normal okhttp dependencies in other libraries. Have you tried changing the okhttp dependency in `android/app/build.gradle` to match the version specified by `react-native-video/android-exoplayer`?

If that doesn't work, you may need to fiddle around with versions or ask on the react-native-video repo, as we can only provide support for the Expo SDK on this repo

**Comment by cheolhow** [comment]
Hi @esamelson .

I posted the issue on the react-native-video repo. The specific error message is the same as the attached image.

https://github.com/react-native-community/react-native-video/issues/1236#issuecomment-429709273

![image](https://user-images.githubusercontent.com/25603134/47423501-22a91880-d7c0-11e8-9456-5392e3a6f398.png)

**Comment by cheolhow** [comment]
https://github.com/react-native-community/react-native-video/issues/1236#issuecomment-429709273

He said Expo is overriding the version of OkHTTP.

**Comment by sjchmiela** [comment]
That is true, we use our own copy of `okhttp` (see the `expolib_v1` package prefix). We'll probably try to remove it as soon as possible to allow better and easier integration between ExpoKit projects and third-party libraries, but I can't promise anything as of now. 🙂

**Comment by cheolhow** [comment]
I hope to be corrected quickly. react-native-video is one of the most popular react-native libraries. It is not good not to be able to use even in the eject expo project.

**Comment by natterstefan** [comment]
It would even be more awesome when expo incorporates [react-native-video](https://github.com/react-native-community/react-native-video#readme) into Expo SDK, not just [enabling us when we eject expo projects](https://docs.expo.io/versions/latest/expokit/eject). Probably, I am just saying this, because I am stuck here https://github.com/expo/expo/issues/2941 still, and do not want to eject yet as the last ressort. Let's see.

**Comment by auchaphon** [comment]
same issue please help.

**Comment by calebomusic** [comment]
Same issue.

In the meantime, recommendations for other 3rd party audio streaming libs that allow backgrounding audio that work w/ ejected expo apps would be appreciated.

**Comment by gsunsnackv** [comment]
Same issue! And we started with the video player inside the Expo project. However it is completely not usable on Android 9.0 . I had to eject and try use react-native-video. But then I run into this. I'm very very stuck and sad ...

**Comment by gsunsnackv** [comment]
> That is true, we use our own copy of `okhttp` (see the `expolib_v1` package prefix). We'll probably try to remove it as soon as possible to allow better and easier integration between ExpoKit projects and third-party libraries, but I can't promise anything as of now. 🙂

@sjchmiela 'as soon as possible' should not have been over 1 year

**Comment by gsunsnackv** [comment]
> > That is true, we use our own copy of `okhttp` (see the `expolib_v1` package prefix). We'll probably try to remove it as soon as possible to allow better and easier integration between ExpoKit projects and third-party libraries, but I can't promise anything as of now. 🙂
> 
> @sjchmiela 'as soon as possible' should not have been over 1 year

My sincere for mis-read the time to thinking it was Jan 2018.

**Comment by sjchmiela** [comment]
https://github.com/expo/expo/pull/3539 unprefixed `okhttp` used in `expoview`, so `react-native-video` should work ok now.

**Comment by yareyaredesuyo** [comment]
Not working in v32.

cannot convert   expolib_v1.okhttp3.OkHttpClient  to   okhttp3.OkHttpClient error detected.

https://github.com/expo/expo/issues/1231

**Comment by sjchmiela** [comment]
Hey, let me rephrase my statement — `react-native-video` should work ok from SDK33 up (which hasn’t been released yet). Sorry for the confusion! 😕

**Comment by yareyaredesuyo** [comment]
OK, thanks for explanation.

It may take time for testing. so I will wait and try v33 and test it. (maybe solve okhttp kind of issues?  It means we can use react-native-video and fast-image without android trap modification. )

And i thought is is better to have a compatibility with react-native-video ( it maybe better than expo implementaion and have background supports. )

**Comment by leninlin** [comment]
@sjchmiela 
How long to wait for the release? Or how can I apply it to version 32?
We really need this fix.

**Comment by djarrin** [comment]
@sjchmiela I am getting the same error here. Is there an estimated time for the SDK33 release? and to @leninlin point is there a work around for this issue?

**Comment by sjchmiela** [comment]
Hey all! Try installing a new version of `expokit` in your project — `32.2.0-alpha.0` (run `yarn add expokit@32.2.0-alpha.0` in your project's directory). It contains the PR that fixed the issue. After upgrading this dependency you'll need to make certain changes to your project:

```diff
diff --git a/android/app/src/main/java/host/exp/exponent/MainApplication.java b/android/app/src/main/
java/host/exp/exponent/MainApplication.java
index a3e694b..7f00092 100644
--- a/android/app/src/main/java/host/exp/exponent/MainApplication.java
+++ b/android/app/src/main/java/host/exp/exponent/MainApplication.java
@@ -30,69 +30,69 @@ import expo.modules.print.PrintPackage;
 import expo.modules.sensors.SensorsPackage;
 import expo.modules.sms.SMSPackage;
 import expo.modules.taskManager.TaskManagerPackage;
-import expolib_v1.okhttp3.OkHttpClient;
+import okhttp3.OkHttpClient;
```

```diff
diff --git a/android/app/build.gradle b/android/app/build.gradle
index 1e8b0c8..5985749 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -162,7 +162,7 @@ dependencies {
     implementation 'com.facebook.soloader:soloader:0.5.1'
 
     implementation 'com.google.android.exoplayer:exoplayer:2.6.1'
-    implementation 'expolib_v1.com.google.android.exoplayer:expolib_v1-extension-okhttp:2.6.1@aar'
+    implementation 'com.google.android.exoplayer:extension-okhttp:2.6.1'
 
     // expo-file-system
     implementation 'com.squareup.okhttp3:okhttp:3.10.0'
```

After applying these changes your ExpoKit at SDK32 will work using unscoped `okhttp`. 🎉

**Comment by vractal** [comment]
Thanks a lot @sjchmiela! I have been trying to build expokit with this fix but with no success so far, so this came  in very handy :)

**Comment by sjchmiela** [comment]
Yeah, many good, important changes happened to `master` since we released SDK32 so it wasn't as easy as cherry-picking and building, so I understand the struggle and relief. 😅

**Comment by leninlin** [comment]
Hi guys!
I temporarily reworked the react-naive-video module for SDK32. Not the latest version (react-native-video), but it works for me.
Welcome...
https://github.com/leninlin/react-native-video/commit/d2665a15579496046b4346b671b8cee8f123a77a

**Comment by djarrin** [comment]
I'm having trouble with this solution (not completely sure this is the correct place to post this, if not let me know and I'll repost in the proper place).

So I followed @sjchmiela instructions. I was moving from expoKit 31 -> expokit 32.2.0-alpha.0 so I was running into this error: `Could not find host.exp.exponent:expoview:31.0.0`: https://github.com/expo/expo/issues/2860
following @sjchmiela advice again I followed the instructions to upgrade android expoKit sdk: https://docs.expo.io/versions/latest/expokit/expokit/#android-1

Once all of this was completed I ran into a dependency conflict:` Android dependency 'com.google.android.exoplayer:extension-okhttp' has different version for the compile (2.6.1) and runtime (2.9.3) classpath. You should manually set the same version via DependencyResolution`

I attempted to solve this by placing this code in my android/build.gradle:
```
subprojects {
    project.configurations.all {
        resolutionStrategy.eachDependency { details ->
            if (details.requested.group == 'com.google.android.exoplayer'
                    && !details.requested.name.contains('extension-okhttp') ) {
                details.useVersion "2.6.1"
            }
        }
    }
    afterEvaluate {project ->
        if (project.hasProperty("android")) {
            android {
                compileSdkVersion 28
                buildToolsVersion "28.0.3"
            }
        }
    }
}
```
but this unfortunately leads to a bunch of errors like these:
![Screen Shot 2019-04-24 at 4 08 07 PM](https://user-images.githubusercontent.com/7516512/56691205-5b617600-66ad-11e9-8473-a52c03491cc6.png)

I'm hoping I'm doing something horribly wrong, any ideas (and please let me know if I should post this somewhere else).

**Comment by adamsythe** [comment]
@djarrin  same error on my side

**Comment by adamsythe** [comment]
@djarrin  
After more fiddling i changed
```
`implementation 'com.google.android.exoplayer:exoplayer:2.6.1'
  implementation 'com.google.android.exoplayer:extension-okhttp:2.6.1'`
```

To

```
 implementation 'com.google.android.exoplayer:exoplayer:2.9.3'
  implementation 'com.google.android.exoplayer:extension-okhttp:2.9.3'
```

I now have a new error

Android dependency 'com.android.support:support-media-compat' has different version for the compile (27.1.1) and runtime (28.0.0) classpath. You should manually set the same version via DependencyResolution

**Comment by djarrin** [comment]
@adamsythe I don't recall all the steps I took to get it to build but instead of manually updating the app/build.gradle file I started using the older version of the android media player. You can find where and how to do this in the README.md (https://github.com/react-native-community/react-native-video) under the android setup and instructions for android/settings.gradle
